packages/hash/frontend/src/components/layout/PageSidebar/AccountEntityTypeList/AccountEntityTypeList.tsx

```diff
@@ -18,132 +109,150 @@ export const AccountEntityTypeList: VFC<AccountEntityTypeListProps> = ({
   accountId,
 }) => {
   const { data } = useAccountEntityTypes(accountId);
-  //   const [order, setOrder] = useState<"asc" | "desc" | undefined>();
-
-  const sortedData = useMemo(() => {
-    return data?.getAccountEntityTypes ?? [];
-  }, [data]);
+  const router = useRouter();
 
-  //   const sortedData = useMemo(() => {
-  //     if (!order) {
-  //       return data?.getAccountEntityTypes ?? [];
-  //     }
+  const [sortType, setSortType] = useState<SortType>("asc");
+  const [searchVisible, setSearchVisible] = useState(false);
+  const [searchQuery, setSearchQuery] = useState("");
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const sortActionsPopupState = usePopupState({
+    variant: "popover",
+    popupId: "type-sort-actions-menu",
+  });
 
-  //     return orderBy(
-  //       data?.getAccountEntityTypes ?? [],
-  //       ["properties.title"],
-  //       [order],
-  //     );
-  //   }, [order, data]);
+  useEffect(() => {
+    if (searchVisible) {
+      searchInputRef.current?.focus();
+    }
+  }, [searchVisible]);
 
-  //   const toggleSort = useCallback(() => {
-  //     if (!order) {
-  //       setOrder("asc");
-  //     }
+  // todo: handle search server side
+  const filteredData = useMemo(() => {
+    let entityTypes = data?.getAccountEntityTypes ?? [];
 
-  //     if (order === "asc") {
-  //       setOrder("desc");
-  //     }
+    if (searchQuery) {
+      entityTypes = entityTypes.filter(({ properties }) =>
+        properties.title.toLowerCase().includes(searchQuery.toLowerCase()),
+      );
+    }
 
-  //     if (order === "desc") {
-  //       setOrder("asc");
-  //     }
-  //   }, [order]);
+    // Right now we just handle ascending/descending and default to ascending
+    // for other sort types
+    return orderBy(
+      entityTypes,
+      ["properties.title"],
+      [sortType === "asc" || sortType === "desc" ? sortType : "asc"],
+    );
+  }, [sortType, data, searchQuery]);
 
   return (
     <Box>
       <NavLink
```

Comment:
Can be addressed separately, but I think there's an inconsistency with how the icons act on hover:

https://user-images.githubusercontent.com/42802102/161054201-10258c6e-201c-4751-baa2-3cee55380bf3.mov

I'm not sure which is desired (if you are as well, worth checking with @MaggieAppleton ), but the icons next to pages only have a distinct background-color when the user hovers over them, whereas for these `NavLink` components the icons have a distinct background colour when the user hovers over any part of the `NavLink` component. My instinct is the former behaviour is desired.